M2/Macaulay2/e/NCAlgebras/FreeAlgebra.hpp

```diff
@@ -0,0 +1,323 @@
+#ifndef _free_algebra_hpp_
+#define _free_algebra_hpp_
+
+#include "Polynomial.hpp"             // for Poly, Monom
+#include "NCAlgebras/FreeMonoid.hpp"  // for FreeMonoid
+#include "NCAlgebras/Word.hpp"        // for Word
+#include "newdelete.hpp"              // for our_new_delete
+#include "ring.hpp"                   // for Ring (ptr only), SumCollector
+#include "ringelem.hpp"               // for ring_elem
+#include "style.hpp"                  // for GEOHEAP_SIZE
+
+#include <gmp.h>                      // for mpz_srcptr, mpq_srcptr
```

Comment:
I must have missed this, but I think `M2/math-include.h` should be used instead.